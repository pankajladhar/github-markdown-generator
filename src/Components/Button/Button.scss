.button {
  -webkit-appearance: none;
  border: none;
  border: 1px #ccc solid;
  box-shadow: rgba(0, 0, 0, .12) 0 1px 6px, rgba(0, 0, 0, .12) 0 1px 4px;
  border-radius: 2px;
  margin-right: 5px;
  font-size: 14px;
  padding: 5px 0;
  text-align: center;
  background: #fff;
  cursor: pointer;

  &.action {
    width: 40px;
  }

  &:hover {
    background-color: rgba(209, 206, 206, .8);
  }

  &:focus {
    outline: none;
  }
}
