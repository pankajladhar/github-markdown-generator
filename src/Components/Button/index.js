import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import './Button.scss';

const propTypes = {
    className: PropTypes.string,
    disabled: PropTypes.bool,
    icon: PropTypes.string,
    name: PropTypes.string,
    handleClick: PropTypes.func.isRequired,
    title: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired
}
const defaultProps = {
    className: "",
    disabled: false
}

export default class Button extends PureComponent {
    render() {
        return (
                <button
                    className={`button ${this.props.className}`.trim()}
                    name={this.props.name} 
                    onClick={this.props.handleClick} 
                    title={this.props.title}
                    disabled={this.props.disabled}
                >
                {this.props.icon && <i className={`fa ${this.props.icon}`} aria-hidden="true"></i>}
                {!this.props.icon && this.props.value}
                </button>
        );
    }
}

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;
