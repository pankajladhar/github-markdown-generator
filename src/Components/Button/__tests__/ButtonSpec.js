import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Button from './../index';

describe('Button', ()=>{
    let buttonProps = {}
    beforeEach(()=>{
        buttonProps = {
            handleClick : ()=>{},
            title: "Button",
            value: "Button"
        }
    });
    
    it('renders correctly when only required props are provided', ()=>{
        let tree = renderer.create(
            <Button  {...buttonProps}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly when only all props are provided', ()=>{
        buttonProps.className="sample";
        buttonProps.name="name";

        let tree = renderer.create(
            <Button  {...buttonProps}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should not render text if icon props is provided', ()=>{
        buttonProps.icon = "fa-bold"
        let tree = renderer.create(
            <Button  {...buttonProps}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree.props.value).toBeUndefined();
    })

    it('should call the provided click handler', ()=>{
        
        let mockHandleClick = jest.fn();
        
        buttonProps.className="sample";
        buttonProps.name="name";
        buttonProps.handleClick = mockHandleClick;
        
        let button = shallow(
            <Button {...buttonProps}/>
        );

        button.simulate('click');
        expect(mockHandleClick).toHaveBeenCalled();
    });

    it('should have disabled property if value disabled props is true ', ()=>{

        buttonProps.disabled=true;
        let tree = renderer.create(
            <Button  {...buttonProps}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree.props.disabled).toBeTruthy();
    })
});
