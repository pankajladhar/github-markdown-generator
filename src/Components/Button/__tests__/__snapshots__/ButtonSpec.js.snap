// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly when only all props are provided 1`] = `
<button
  className="button sample"
  disabled={false}
  name="name"
  onClick={[Function]}
  title="Button"
>
  Button
</button>
`;

exports[`Button renders correctly when only required props are provided 1`] = `
<button
  className="button"
  disabled={false}
  name={undefined}
  onClick={[Function]}
  title="Button"
>
  Button
</button>
`;

exports[`Button should have disabled property if value disabled props is true  1`] = `
<button
  className="button"
  disabled={true}
  name={undefined}
  onClick={[Function]}
  title="Button"
>
  Button
</button>
`;

exports[`Button should not render text if icon props is provided 1`] = `
<button
  className="button"
  disabled={false}
  name={undefined}
  onClick={[Function]}
  title="Button"
>
  <i
    aria-hidden="true"
    className="fa fa-bold"
  />
</button>
`;
